File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def solution(today, terms, privacies):
    answer = []
    terms_dict  = {terms[i].split()[0] : terms[i].split()[1] for i in range(0, len(terms))}

    y,m,d = today.split('.')
    today = int(y)*12*28 + int(m)*28 + int(d)
    
    for i, value in enumerate(privacies):
        date, key = value.split()
        if key in list(terms_dict.keys()):
            val = terms_dict[key]
            print(val)
            
            yy, mm, dd = date.split('.')
            date = int(yy)*12*28 + int(mm)*28 + int(dd) + int(val)*28 - 1
            if date < today:
                answer.append(i+1)

                
                
    return answer

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;
int cal(string today) {
    string year = today.substr(0, 4);
    string month = today.substr(5, 2);
    string day = today.substr(8, 2);
    return stoi(year)*12*28 + stoi(month)*28 + stoi(day);
}

vector<int> solution(string today, vector<string> terms, vector<string> privacies) {
    vector<int> answer;
    unordered_map<char, int> t;
    int todays = cal(today);

    for (int i=0; i<terms.size(); i++) {
        t[terms[i][0]] = stoi(terms[i].substr(2, terms[i].size()-2))*28;
    }
    
    for (int i=0; i<privacies.size(); i++) {
        char type = privacies[i][11];
        int days = cal(privacies[i]) + t[type] -1;
        if (todays > days) answer.push_back(i+1);
    }
    return answer;
}